### ISSUE TITLE
[android] Do not ask to enable location services if they're disabled in system settings

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#4390]
Now if location services are disabled, OM displays the crossed crosshair location button and asks to enable them on the first launch (under the hood OM starts detecting the user position, assuming that the user launched the map to see his/her actual position on it). And then does not ask any more if OM was in the background for less than 30 hours.

Some users are using OM with Location Services disabled. And those users are annoyed by that dialog appearing in 30+ hours after moving OM into the background.

A possible fix would be to avoid activating positioning in 30+ hours if two of the below conditions are met:
- Location Service is disabled in the system
- OM was moved into the background with a location mode "Off", e.g. the crosshair button was crossed.

**Comment by rtsisyk** [comment]
This proposal is controversial. People will not get position in the app, delete the app and give * star in the store. The current Android behavior already pessimized the most common use case when location is enabled and should just work of the box. This creates inconvenience for the majority of users who just want to use the app for their daily tasks. iOS version works better and location is just always enabled no matter what. This is how other apps work as well. 

Please elaborate on this proposal to cover other uses-cases and needs of the majority of users.

**Comment by biodranik** [comment]
@fusionneur doesn't popup offer to enable location now?

**Comment by nach0** [comment]
I noticed that in the latest update (2023.09.22-27-F-Droid) and also the beta 2023.09.25-17-FDroid-beta (downloaded from #3130) this issue is back.

The pop-up asking me to "Enable Location Services" was gone for months now (I don't recall the exact update when this was fixed), but today I am getting this pop-up every time that I launch Organic Maps (any of the 2 versions mentioned above). If I launch the app, dismiss the message, then close it and instantly launch it again the pop-up is there again. I cleared all apps in the App Drawer and launched Organic Maps again, same behaviour. I restarted my phone, same behaviour.

I only noticed this today, so I believe the issue was not present in these versions on first launch when I had location services turned off. Yesterday I used both apps with location services activated, and today I started noticing this behaviour. Therefore, it could be that using location services yesterday triggered this behaviour, which I didn't see before using the apps with location services off.
